### ISSUE TITLE
"Export as..." does not mark the tmx file as changed

### ISSUE THREAD

**Issue by thomasgoldstein** [mapeditor/tiled#2940]
Steps:
- Create or open a tmx file
- File / Export as...
- Save the export file somewhere

Actual result: the tmx file is **not** marked as changed (*). Trying to close the tmx file does **not** trigger a prompt asking you if you want to save the pending changes (the persistence of the `<export target="[path]" format="[ext]"/>` element).

Expected result: the tmx file is marked as changed (*). Trying to close the tmx file triggers a prompt asking you if you want to save the pending changes (the persistence of the `<export target="[path]" format="[ext]"/>` element).

**Comment by bjorn** [comment]
Right, this is a result of the remembering of the export format not being a change you can undo.

I wonder if saving it into the map file is the best approach anyway. I implemented it like this for Tiled 1.3, but since Tiled 1.4 we have projects & sessions. Remembering of the last used export location and format could be moved into the session (which is saved automatically), and the project could have a general setup for automated export for all maps. Does that sound reasonable?

**Comment by thomasgoldstein** [comment]
Thanks for your answer. I think you know best what's good for the editor. I don't have an opinion on this topic because I've never used projects or sessions (I didn't feel the need to, as I was happy with the maps alone). But if you move the last export to projects and sessions, then I might start using them.

**Comment by eishiya** [comment]
I support moving the export format and location into project/session data and removing it from maps.

My situation is perhaps unusual in that, in answering questions on the forum and Discord, I often experiment with different export formats and settings using the files I have available, and then I have to deal with git thinking the original files changed because I... saved a different file...? I don't think any kind of export should modify the original file in any way, period.
